src/storage/hummock_sdk/src/compaction_group/hummock_version_ext.rs

```diff
@@ -604,6 +570,94 @@ pub fn build_initial_compaction_group_levels(
     }
 }
 
+pub fn check_sst_id_exist(
+    sst_id: HummockSstableId,
+    level: &mut Level,
+    move_table_ids: &[u32],
+) -> bool {
+    if level.level_type == LevelType::Overlapping as i32 {
+        if let Some(ret) = level
+            .table_infos
+            .iter_mut()
+            .find(|sst| sst.sst_id == sst_id)
+        {
+            ret.table_ids.extend(move_table_ids);
+            return true;
+        }
+    } else if let Ok(idx) = level
+        .table_infos
+        .binary_search_by(|sst| sst.sst_id.cmp(&sst_id))
+    {
+        level.table_infos[idx].table_ids.extend(move_table_ids);
+        return true;
+    }
+    false
+}
+
+fn split_sst_info_for_level(
+    level: &mut Level,
+    split_id_vers: &mut Vec<SstSplitInfo>,
+    member_table_ids: &HashSet<u32>,
+    new_sst_id: &mut u64,
+) -> Vec<SstableInfo> {
+    // Remove SST from sub level may result in empty sub level. It will be purged
+    // whenever another compaction task is finished.
+    let mut removed = vec![];
+    let mut insert_table_infos = vec![];
+    for sst_info in &mut level.table_infos {
+        let removed_table_ids = sst_info
+            .table_ids
+            .iter()
+            .filter(|table_id| member_table_ids.contains(table_id))
+            .cloned()
+            .collect_vec();
+        if !removed_table_ids.is_empty() {
+            let is_trivial = removed_table_ids.len() == sst_info.table_ids.len();
+            let mut branch_table_info = sst_info.clone();
+            branch_table_info.sst_id = *new_sst_id;
+            *new_sst_id += 1;
+            let parent_old_sst_id = sst_info.get_sst_id();
+            split_id_vers.push((
+                branch_table_info.get_object_id(),
+                branch_table_info.get_sst_id(),
+                parent_old_sst_id,
+                if is_trivial {
+                    None
+                } else {
+                    sst_info.sst_id = *new_sst_id;
+                    *new_sst_id += 1;
+                    Some(sst_info.get_sst_id())
+                },
+            ));
+            branch_table_info.table_ids = removed_table_ids;
```

Comment:
??? It's not consistent with the definition in `commit_epoch`.

Reply:
It is a mistake. We would not change table ids when split